lib/mauth/rack.rb

```diff
@@ -70,11 +84,20 @@ def should_authenticate?(env)
       end
     end
 
-    # signs outgoing responses
+    # signs outgoing responses with only the protocol used to sign the request.
     class ResponseSigner < MAuth::Middleware
       def call(env)
+        # QUESTION does instaniate a new request object, rather than doing a check for the headers here, add too much overhead?
```

Comment:
I think once #24  is merged, it should be expanded to compare v1 and v2 authorization and signing. 
Also by running it on develop vs this branch you could check if performance degraded